### DISCUSSION TITLE
Feature request: Move & Resize background windows

### DISCUSSION THREAD

**Discussion by johnniewilcox** [rxhanson/RectanglePro-Community#94] [General]
Just discovered Rectangle Pro. Fantastic UX. Thank you for this software.

I have ~~a couple of feature requests~~ one feature request.

1. ~~Allow Rectangle Pro to move / resize background windows.~~
2. Add option for (custom) modifier key to bring hovered window to front.

Thank you for considering.

**Comment by rxhanson** [comment]
Thanks for using Pro! Sorry for the late reply.

You can disable the automatic 'bring to front' for move/resize with the following terminal command followed by an app restart:

`defaults write com.knollsoft.Hookshot windowToFront -int 1`

(The default value is 0)

I previously had the default behavior of allowing the additional press of the fn key to keep the background window in the background (the inverse of what you're looking for), but I think I abandoned that when I started allowing the fn key to be used as a standalone modifier. 

I'll see if I can roll your request in when I'm back in that part of the code again.

**Reply by Harrisonbro** [reply]
Thank you! I too came to Rectangle Pro from Zooom so wanted this. 

Thanks for your work on the app 👏 Any Mac without Rectangle feels utterly broken

**Comment by johnniewilcox** [comment]
Thank you for providing `windowToFront` the hidden preference for Rectangle Pro!

I have literally been waiting for nearly five years (since Mojave's release) for Rectangle Pro's feature set, specifically the ability to move / resize background windows.

Rectangle Pro's highly-polished UI and UX (UIX) do not merely replicate affordances of Golden Age Mac OS applications. Rectangle Pro establishes a modern 21st-century vernacular for macOS app UIX that leverages both keystrokes and gestures in intuitive, empowering, and delightful ways.

Thank you so much for all your hard work with this app. The quality will be obvious to anyone who cares to look.

A little bit of background follows regarding my remaining feature request to "Add an option for a (custom) modifier key to bring hovered window to front."

I have been a long-time user of [Zooom](https://zooom.en.softonic.com/mac) which, alas, is moribund. That app separates move / resize from focus (Auto Raise) as you can see in this screen shot.

![Untitled](https://user-images.githubusercontent.com/20446167/162224102-41e1434c-c967-4a8c-897a-9b08162089c5.png)

Today, I am retiring Zooom and moving forward with Rectangle Pro. For now, every time I unintentionally throw a window to the `top | bottom | left | right` half of my monitor, I have a quick laugh at my nearly 20-year muscle memory and also my great fortune in having discovered Rectangle Pro.

**Comment by rxhanson** [comment]
Thanks for the kind words! I'm glad you are enjoying it.

Also, thanks for the screenshot. I'll plan on adding these Auto Raise settings, since this make sense to me, but I think it's going to be some time before I get to it. I have two large features that are next in my queue, and after each large feature I typically make rounds on some of my other apps in between.

I've had I think two other previous Zooom users looking for some functionality from that app. That snapping grid thing in it seemed pretty cool.

**Comment by awgneo** [comment]
Is this issue related to the ability to unhide windows that are hidden in the dock when restoring an application layout? I have a layout set to launch all applications if they aren't already open. However, if I hide/minimize one of the windows of this group and re-issue the layout shortcut, it doesn't unhide the window to its proper location. Perhaps this can be a setting? Thanks :)

**Comment by rxhanson** [comment]
@awgneo, this is not related. I made a separate issue to track your request. https://github.com/rxhanson/RectanglePro-Community/issues/72
